include/icubFloatingBaseEstimatorV1.h

```diff
@@ -0,0 +1,632 @@
+/*
+################################################################################
+#                                                                              #
+# Copyright (C) 2018 Fondazione Istituto Italiano di Tecnologia (IIT)          #
+# All Rights Reserved.                                                         #
+#                                                                              #
+################################################################################
+
+# @authors: Prashanth Ramadoss <prashanth.ramadoss@iit.it>
+#           Giulio Romualdi    <giulio.romualdi@iit.it>
+#           Silvio Traversaro  <silvio.traversaro@iit.it>
+#           Daniele Pucci      <daniele.pucci@iit.it>
+*/
+
+#ifndef ICUB_FLOATING_BASE_ESTIMATOR_V1_H
+#define ICUB_FLOATING_BASE_ESTIMATOR_V1_H
+
+#include <iDynTree/Estimation/AttitudeMahonyFilter.h>
+#include <iDynTree/Estimation/SimpleLeggedOdometry.h>
+#include <iDynTree/Estimation/BipedFootContactClassifier.h>
+#include <iDynTree/Estimation/AttitudeQuaternionEKF.h>
+
+#include <iDynTree/Model/Model.h>
+#include <iDynTree/KinDynComputations.h>
+#include <iDynTree/ModelIO/ModelLoader.h>
+#include <iDynTree/yarp/YARPConversions.h>
+#include <iDynTree/Core//EigenHelpers.h>
+
+#include <yarp/os/PeriodicThread.h>
+#include <yarp/os/ResourceFinder.h>
+#include <yarp/os/RpcClient.h>
+#include <yarp/dev/DeviceDriver.h>
+#include <yarp/dev/Wrapper.h>
+#include <yarp/dev/IEncoders.h>
+#include <yarp/dev/IAnalogSensor.h>
+#include <yarp/dev/GenericSensorInterfaces.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/LockGuard.h>
+#include <yarp/eigen/Eigen.h>
+#include <yarp/sig/Vector.h>
+#include <yarp/sig/Matrix.h>
+#include <yarp/dev/IFrameTransform.h>
+#include <memory>
+#include <string>
+#include <WalkingLogger.hpp>
+#include <thrifts/floatingBaseEstimationRPC.h>
+
+#include <iCub/ctrl/filters.h>
+
+inline double deg2rad(const double angleInDeg)
+{
+    return angleInDeg*M_PI/180.0;
+}
+
+inline double rad2deg(const double angleInRad)
+{
+    return angleInRad*180.0/M_PI;
+}
+
+inline void convertVectorFromDegreesToRadians(iDynTree::VectorDynSize & vector)
+{
+    for(size_t i=0; i < vector.size(); i++)
+    {
+        vector(i) = deg2rad(vector(i));
+    }
+
+    return;
+}
+
+namespace yarp {
+    namespace dev {
+
+
+    class icubFloatingBaseEstimatorV1 : public yarp::dev::DeviceDriver,
```

Comment:
It would be great if we could avoid explicit reference to `iCub` in the device name, in the end even if there are a few icub-ism I think it is  quite a general legged robot estimator, so a more general name could come handy for the future.

Reply:
Addressed in https://github.com/robotology/whole-body-estimators/pull/1/commits/714c6ccc684abcde00050378b2ddb09d6054ddd9